apps/omg_api/lib/block_queue.ex

```diff
@@ -62,6 +62,10 @@ defmodule OMG.API.BlockQueue do
     end
 
     def init(:ok) do
+      {:ok, %{}, {:continue, :setup}}
+    end
+
+    def handle_continue(:setup, %{}) do
```

Comment:
I think behavior like that is desired as it decouples the application from the service.

Reply:
Sorry for previous comment. Somehow I've detected the problem but managed to misdiagnosed it. The actual problem is - it's a regression leading to following scenario being possible: `watcher is starting amidst mass exit and is reporting green because its geth instance is stuck processing a week old blocks`.

Reply:
Hm... alright, but that ought to fix itself, right? The gist of the code didn't change. The process would start, decoupled from the supervisor, once the setup starts, it would crash. Crashing will bring the service down if it can't get into the desired state. 

Is there a test that would validate the assumptions I've made?

Reply:
Misdiagnosing? I did that twice. It's a block queue we are talking about, not watcher. :sigh:

BlockQueue needs geth to be synchronized, and the check will make sure it will crash if geth is not synced on start.
And you are right that after a series of restarts it will bring the whole thing down. But we will be accepting traffic on the `OMG.RPC.Web.Endpoint`, while not being able to submit any new blocks.

Reply:
Ok. I think I understand! Thanks, I'll ping you once I fix this.